File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int c=0; long long s=987654345678;
        long long res=0;
        int n=matrix.size();
        for(int i=0;i<n;++i)
            for(int j=0;j<n;++j){
                int r=matrix[i][j];
                s=min(s,(long long)abs(r));
                if(r<0)c++;
                res+=abs(r);
            }
        if(c%2==0) return res;
        return res-(s*2);
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        res=0
        s=987654567898765
        c=0
        for i in range(len(matrix)):
            for j in range(len(matrix)):
                r=matrix[i][j]
                s=min(s,abs(r))
                if r<0:
                    c+=1
                res+=abs(r)
        if c%2==0: return res
        else:
            return res-s*2